field/arena.go

```diff
@@ -836,22 +845,20 @@ func (arena *Arena) handlePlcOutput() {
 	}
 }
 
-func (arena *Arena) handleEstop(station string, state bool) {
+func (arena *Arena) handleTeamStop(station string, eStopState, aStopState bool) {
 	allianceStation := arena.AllianceStations[station]
-	if state {
-		if arena.MatchState == AutoPeriod {
-			allianceStation.Astop = true
-		} else {
-			allianceStation.Estop = true
-		}
-	} else {
-		if arena.MatchState != AutoPeriod {
-			allianceStation.Astop = false
-		}
-		if arena.MatchTimeSec() == 0 {
-			// Don't reset the e-stop while a match is in progress.
-			allianceStation.Estop = false
-		}
+	if eStopState {
+		allianceStation.Estop = true
+	} else if arena.MatchTimeSec() == 0 {
+		// Keep the E-stop latched until the match is over.
+		allianceStation.Estop = false
+	}
+	if aStopState {
+		allianceStation.Astop = true
+	} else if arena.MatchState != AutoPeriod {
+		// Keep the A-stop latched until the autonomous period is over.
```

Comment:
this is good, coz in hardware our A-stop is a momentary button while the E-stop is a locking button which needs to be manually released